src/snyk/common/commands/commandController.ts

```diff
@@ -0,0 +1,106 @@
+import _ from 'lodash';
+import * as vscode from 'vscode';
+import { ISnykCodeService } from '../../snykCode/codeService';
+import { severityAsText } from '../../snykCode/utils/analysisUtils';
+import { createDCIgnore } from '../../snykCode/utils/ignoreFileUtils';
+import { CodeIssueCommandArg } from '../../snykCode/views/interfaces';
+import { OssService } from '../../snykOss/services/ossService';
+import { OssIssueCommandArg } from '../../snykOss/views/vulnerabilityProvider';
+import { analytics } from '../analytics/analytics';
+import {
+  SNYK_COPY_AUTH_LINK_COMMAND,
+  SNYK_OPEN_BROWSER_COMMAND,
+  SNYK_OPEN_LOCAL_COMMAND,
+  VSCODE_GO_TO_SETTINGS_COMMAND,
+} from '../constants/commands';
+import { COMMAND_DEBOUNCE_INTERVAL, IDE_NAME, SNYK_NAME_EXTENSION, SNYK_PUBLISHER } from '../constants/general';
+import { IOpenerService } from '../services/openerService';
+import { OpenIssueCommandArg, OpenCommandIssueType } from './types';
+
+export class CommandController {
+  private debouncedCommands: Record<string, _.DebouncedFunc<(...args: unknown[]) => Promise<unknown>>> = {};
+
+  constructor(
+    private openerService: IOpenerService,
+    private snykCode: ISnykCodeService,
+    private ossService: OssService,
+  ) {}
+
+  openBrowser(url: string): unknown {
+    return this.executeCommand(SNYK_OPEN_BROWSER_COMMAND, this.openerService.openBrowserUrl.bind(this), url);
+  }
+
+  copyAuthLink(): unknown {
+    return this.executeCommand(SNYK_COPY_AUTH_LINK_COMMAND, this.openerService.copyOpenedUrl.bind(this.openerService));
+  }
+
+  openLocal(path: vscode.Uri, range?: vscode.Range): void {
+    // todo: add error reporting
+    void vscode.window.showTextDocument(path, { viewColumn: vscode.ViewColumn.One, selection: range });
+  }
+
+  openSettings(): void {
+    void vscode.commands.executeCommand(VSCODE_GO_TO_SETTINGS_COMMAND, `@ext:${SNYK_PUBLISHER}.${SNYK_NAME_EXTENSION}`);
+  }
+
+  async createDCIgnore(custom = false, path?: string): Promise<void> {
```

Comment:
@michelkaporin I'm trying to understand the logic in this class. Not sure if I got it right. But maybe this functionality related to some other part/class?

Reply:
@aldanchenko This class is an entry point for commands issued from different places within extension. Command is an instrument to trigger some actions within VS Code. They're defined in package.json with bindings that map to this class. More info: https://code.visualstudio.com/api/extension-guides/command

I've extracted existing code that used to reside next to various bindings (snyk/extension.ts) into a separate class to encapsulate it better.